beacon_node/network/src/sync/backfill_sync/mod.rs

```diff
@@ -253,35 +250,21 @@ impl<T: BeaconChainTypes> BackFillSync<T> {
                 self.set_state(BackFillState::Syncing);
 
                 // Obtain a new start slot, from the beacon chain and handle possible errors.
-                match self.reset_start_epoch() {
-                    Err(ResetEpochError::SyncCompleted) => {
-                        error!(self.log, "Backfill sync completed whilst in failed status");
-                        self.set_state(BackFillState::Completed);
-                        return Err(BackFillError::InvalidSyncState(String::from(
-                            "chain completed",
-                        )));
-                    }
-                    Err(ResetEpochError::NotRequired) => {
-                        error!(
-                            self.log,
-                            "Backfill sync not required whilst in failed status"
-                        );
-                        self.set_state(BackFillState::NotRequired);
-                        return Err(BackFillError::InvalidSyncState(String::from(
-                            "backfill not required",
-                        )));
-                    }
-                    Ok(_) => {}
+                if let Err(e) = self.reset_start_epoch() {
+                    let ResetEpochError::SyncCompleted = e;
```

Comment:
This line seems redundant?

Reply:
This is to make sure we handle all variants if the `ResetEpochError` gets refactored.

If we wrote:

```rust
if let Err(ResetEpochError::SyncCompleted) = self.reset_start_epoch() {
   ...
}
```

Then we could accidentally _not_ run this code for a newly added variant. Similarly if we just use `e` and run this code for all variants then that may not be correct. So we're forcing a breakage if any variants get added in future, similar to if we used `match`, but with less nesting:

```rust
match self.reset_start_epoch() {
    Err(ResetEpochError::SyncCompleted) => { ... },
    Ok(_) => {},
}
```

I'll add a comment.

Reply:
Comment added https://github.com/sigp/lighthouse/pull/5978/commits/c4d6ef3544b529bbbc8a919916165e6e976eb1fb